crates/solver/src/liquidity/order_converter.rs

```diff
@@ -100,14 +103,15 @@ impl SettlementHandling<LimitOrder> for OrderSettlementHandler {
         execution: LimitOrderExecution,
         encoder: &mut SettlementEncoder,
     ) -> Result<()> {
-        let is_native_token_buy_order = self.order.data.buy_token == BUY_ETH_ADDRESS;
-        if is_native_token_buy_order {
+        let manage_native_token_unwraps =
+            self.order.data.buy_token == BUY_ETH_ADDRESS && self.manage_native_token_unwraps;
```

Comment:
This will need to happen for #2215 as well. Do solvers really want to change that logic?

Reply:
@fleupold what do you exactly mean? doe I need to alter the logic of #2215 ? :thinking: should I skip the wrap() done [here](https://github.com/cowprotocol/services/pull/2661/files#diff-4c8f610bdbea68810a9395f71782c8a9fb5a6931e9e814a2f5f9b33bdc847e2eR371)?

Reply:
https://github.com/cowprotocol/services/pull/2670/files#diff-16555bd09cf8f3fcb161d4a75ef57f79168ad2ed63f29f80b4439ae222d3517bR197 needs to adopt this logic

Reply:
Oh I see, thanks @fleupold ! Added!